plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py

```diff
@@ -0,0 +1,317 @@
+import random
+from os import _exit
+from os.path import exists
+
+lock = Lock()
+
+PHY_RAW_ERROR_LANE0 = "phy_raw_errors_lane0"
+PHY_RAW_ERROR_LANE1 = "phy_raw_errors_lane1"
+PHY_RAW_ERROR_LANE2 = "phy_raw_errors_lane2"
+PHY_RAW_ERROR_LANE3 = "phy_raw_errors_lane3"
+PHY_EFF_ERROR = "phy_effective_errors"
+PHY_SYMBOL_ERROR = "phy_symbol_errors"
+RCV_PACKETS_COUNTER = "PortRcvPktsExtended"
+RCV_ERRORS_COUNTER = "PortRcvErrorsExtended"
+RCV_REMOTE_PHY_ERROR_COUNTER = "PortRcvRemotePhysicalErrorsExtended"
+TEMP_COUNTER = "CableInfo.Temperature"
+FEC_MODE = "fec_mode_active"
+ENDPOINT_CONFIG = {}
+
+class CsvEndpointHandler(BaseHTTPRequestHandler):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def do_GET(self):
+        ENDPOINT_CONFIG["ITERATION_TIME"] += 1
+        self.send_response(200)
+        self.send_header('Content-type', 'text/plain')
+        self.end_headers()
+
+        path = self.path
+        if path not in ENDPOINT_CONFIG:
+            self.wfile.write(b'Endpoint not found')
+            return
+
+        endpoint = ENDPOINT_CONFIG[path]
+        data = endpoint['data']
+        self.wfile.write(data.encode())
+
+DIFFERENT_DEFAULT_VALUES = {
+    # because the plugin reads the meta data to know the first temperature and we cannot stream the metadata.
+    TEMP_COUNTER:"5",
+    RCV_PACKETS_COUNTER:"10000000",
+    PHY_RAW_ERROR_LANE0:0.01,
+    PHY_RAW_ERROR_LANE1:0.01,
+    PHY_RAW_ERROR_LANE2:0.01,
+    PHY_RAW_ERROR_LANE3:0.01,
+}
+
+ALL_DATA_TEST = {
+    # all positive tests
+    # iteration, row index, counter name = value
+    (1,0,PHY_SYMBOL_ERROR):0, # example
+    (1,3,RCV_ERRORS_COUNTER):50,
+    # testing packet drop rate criteria
+    (2,3,RCV_ERRORS_COUNTER):500,
+
+    # testing temperature changes
+    (3,4,TEMP_COUNTER):90,
+    # testing temperature max difference
+    (3,6,TEMP_COUNTER):25,
+
+    (4,8,RCV_REMOTE_PHY_ERROR_COUNTER):50,
+    # testing packet drop rate criteria from the second counter. because we look on rate
+    (5,8,RCV_REMOTE_PHY_ERROR_COUNTER):500,
+    
+    # testing ber calculation
+    (0,2,PHY_RAW_ERROR_LANE0):0.001,
+    (0,2,PHY_RAW_ERROR_LANE1):0.001,
+    (0,2,PHY_RAW_ERROR_LANE2):0.001,
+    (0,2,PHY_RAW_ERROR_LANE3):0.001,
+
+    # testing ber calculation rate it is so high because we try to do it instead of 25 minutes, now.
+    (9,2,PHY_RAW_ERROR_LANE0):1024**5,
+    (9,2,PHY_RAW_ERROR_LANE1):1024**6,
+    (9,2,PHY_RAW_ERROR_LANE2):1024**5,
+    (9,2,PHY_RAW_ERROR_LANE3):1024**5,
+    
+    # should not work now, only after 30 (3 iterations) seconds, which is 5 more iterations
+    
+
+    # negative tests
+    # testing ber calculation ( should not pass as not all are not equal to 0)
+    (8,0,PHY_RAW_ERROR_LANE1):0.001,
+    (8,0,PHY_RAW_ERROR_LANE2):0.001,
+    (8,0,PHY_RAW_ERROR_LANE3):0.001,
+
+    (9,0,PHY_RAW_ERROR_LANE1):1024**6,
+    (9,0,PHY_RAW_ERROR_LANE2):1024**6,
+    (9,0,PHY_RAW_ERROR_LANE3):1024**6,
+
```

Comment:
it looks like link downed event was not added, please update

Reply:
added link down event